core/src/main/scala/org/http4s/headers/Range.scala

```diff
@@ -78,7 +78,7 @@ object Range extends HeaderKey.Internal[Range] with HeaderKey.Singleton {
     val suffixByteRangeSpec = negativeLong.map(SubRange(_))
 
     // byte-range-set  = 1#( byte-range-spec / suffix-byte-range-spec )
-    val byteRangeSet = Rfc7230.headerRep1(byteRangeSpec.backtrack.orElse1(suffixByteRangeSpec))
+    val byteRangeSet = Rfc7230.headerRep1(byteRangeSpec.orElse1(suffixByteRangeSpec))
 
```

Comment:
In an earlier implementation of `byteRangeSpec` I was incorrectly accepting `-` as the first character and `.backtrack` made `byteRangeSet` work. Now the first character of `byteRangeSpec` is a digit and for `suffixByteRangeSpec` is `-` so presumably the correct parser can be chosen without backtracking.

Reply:
That sounds correct to me.